daemon/graphdriver/overlay2/overlay.go

```diff
@@ -312,23 +322,33 @@ func (d *Driver) Cleanup() error {
 // CreateReadWrite creates a layer that is writable for use as a container
 // file system.
 func (d *Driver) CreateReadWrite(id, parent string, opts *graphdriver.CreateOpts) error {
+	if opts != nil && len(opts.StorageOpt) != 0 && !projectQuotaSupported {
+		return fmt.Errorf("--storage-opt is supported only for overlay over xfs with 'pquota' mount option")
+	}
+
+	if opts == nil {
+		opts = &graphdriver.CreateOpts{}
+	}
+
+	if len(opts.StorageOpt) == 0 {
+		opts.StorageOpt = map[string]string{
+			"size": fmt.Sprintf("%d", d.options.quota.Size),
+		}
+	}
 	return d.Create(id, parent, opts)
 }
 
 // Create is used to create the upper, lower, and merge directories required for overlay fs for a given id.
 // The parent filesystem is used to configure these directories for the overlay.
 func (d *Driver) Create(id, parent string, opts *graphdriver.CreateOpts) (retErr error) {
 
-	if opts != nil && len(opts.StorageOpt) != 0 && !projectQuotaSupported {
-		return fmt.Errorf("--storage-opt is supported only for overlay over xfs with 'pquota' mount option")
-	}
-
```

Comment:
IF you would have moved the code of setting pquota to CreateReadWrite() moving the test would have made sense, but the pquota code stayed in this function (for a good reason) so the "assert" test should stay here as well and then we don't need to ASSUME that Create() cannot be called with storage options except from CreateReadWrite().